.github/PULL_REQUEST_TEMPLATE/pull_request.md

```diff
@@ -0,0 +1,13 @@
+## Description
+
+Closes # (issue)
+
+## Reviewer Notes
+
+<!--- Provide any notes to the reviewer that might help them more easily
+      understand the changeset. --->
+
+## Test Instructions
+
+<!--- Provide instructions to the reviewer on how to explicitly test
+      these changes. --->
```

Comment:
[question] Did we talk about doing something like the Conan center one, where this could have checkboxes such as 

- [ ] User documentation has been updated
- [ ] RELEASE_NOTES have been updated

etc...